### PR TITLE
Sl 159

### PR DESCRIPTION
**Pull request by RS882**
refactor: Setting permissions for execution of saveNewImage and deleteImage methods in ImageService

### CODE REVIEW

File: src/main/java/de/ait/secondlife/controllers/CategoryController.java [hunk 1]
```diff
@@ -84,6 +87,7 @@ public ResponseEntity<List<CategoryDto>> getAllPlusHidden() {
                     content = @Content(mediaType = "application/json", schema = @Schema(implementation = ResponseMessageDto.class)))}
     )
     public ResponseEntity<CategoryDto> add(@Valid @RequestBody CategoryCreationDto dto) {
+        utilities.checkCredentialsToConnectImageToEntity(dto.getBaseNameOfImages());
```
Comment (ychepel) [inline src/main/java/de/ait/secondlife/controllers/CategoryController.java:90]: I would move all these checkings to some service class. First, this is a business logic. Second, for checking you use another service classes

File: src/main/java/de/ait/secondlife/services/utilities/EntityUtilities.java [hunk 2]
```diff
@@ -1,19 +1,34 @@
...
 
+import javax.security.auth.login.CredentialException;
+import java.util.HashSet;
+import java.util.Set;
+
 
 @Component
 @RequiredArgsConstructor
-public class EntityUtilities {
+public class EntityUtilities implements ImageConstants {
```
Comment (ychepel) [inline src/main/java/de/ait/secondlife/services/utilities/EntityUtilities.java:23]: IMHO, the solution with this utility class and its methods is not very obvious. I'd like to discuss

File: src/main/java/de/ait/secondlife/controllers/CategoryController.java (hunk 1 above)
Comment (RS882) [reply src/main/java/de/ait/secondlife/controllers/CategoryController.java]: fixed

File: src/main/java/de/ait/secondlife/services/utilities/EntityUtilities.java (hunk 2 above)
Comment (RS882) [reply src/main/java/de/ait/secondlife/services/utilities/EntityUtilities.java]: fixed

**Review by ychepel — APPROVED** [review]

**Pull request merged**